arrow-libs/core/arrow-core/src/jvmMain/kotlin/arrow/core/raise/CancellationExceptionNoTrace.kt

```diff
@@ -6,7 +6,11 @@ import kotlin.coroutines.cancellation.CancellationException
  * Inspired by KotlinX Coroutines:
  * https://github.com/Kotlin/kotlinx.coroutines/blob/3788889ddfd2bcfedbff1bbca10ee56039e024a2/kotlinx-coroutines-core/jvm/src/Exceptions.kt#L29
  */
-internal actual open class CancellationExceptionNoTrace : CancellationException(RaiseCancellationExceptionCaptured) {
+@Suppress(
+  "EXPECT_ACTUAL_CLASSIFIERS_ARE_IN_BETA_WARNING",
+  "SEALED_INHERITOR_IN_DIFFERENT_MODULE"
+)
+internal actual class NoTrace actual constructor(raised: Any?, raise: Raise<Any?>) : RaiseCancellationException(raised, raise) {
```

Comment:
I am wondering whether we really need to support Android <= 6.0. Otherwise we can simplify this in a common module, instead of requiring platform-dependent implementations.

Reply:
We need this to be in `jvmMain` because `fillInStacktrace` is JVM-specific.

IIRC, the `stackTrace = emptyArray()` is needed in addition to disabling stacktrace creation for Android. Otherwise Android 6.0, and below will crash if the `stackTrace` is not `emptyArray` but was not created. (This is stolen from KotlinX, https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-core/jvm/src/Exceptions.kt#L43).

So removing that will still not allow us to disable stack traces in common.